### PR TITLE
refactor(rust): use typestate pattern on `ImportRecord`

### PR DESCRIPTION
**Pull request by hyf0**
### Description

Typestate pattern would be great solution on solving having too many one-time-initialized option fields on structs.

### CODE REVIEW

**Comment by hyf0** [comment]
* **#2374** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2374?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2373** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2373?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @hyf0 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/rolldown_common/src/types/import_record.rs [hunk 1]
```diff
@@ -33,19 +36,44 @@ bitflags::bitflags! {
...
 }
 
+#[derive(Debug)]
+pub struct ImportRecord<State: Debug> {
+  pub state: State,
+  pub module_request: Rstr,
+  pub kind: ImportKind,
+  pub namespace_ref: SymbolRef,
+  pub meta: ImportRecordMeta,
+}
```
Comment (IWANABETHATGUY) [inline crates/rolldown_common/src/types/import_record.rs:49]: Can you preserve the comments?

Comment (hyf0) [reply crates/rolldown_common/src/types/import_record.rs:49]: sure

**Comment by hyf0** [comment]
### Merge activity

* **Oct 5, 3:07 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/rolldown/rolldown/2374) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

**Pull request merged**